powertools-idempotency/powertools-idempotency-core/src/main/java/software/amazon/lambda/powertools/idempotency/persistence/BasePersistenceStore.java

```diff
@@ -339,7 +339,7 @@ private MessageDigest getHashAlgorithm() {
     private void validatePayload(JsonNode data, DataRecord dataRecord) throws IdempotencyValidationException {
         if (payloadValidationEnabled) {
             String dataHash = getHashedPayload(data);
-            if (!StringUtils.equals(dataHash, dataRecord.getPayloadHash())) {
+            if (!isEqual(dataRecord.getPayloadHash(), dataHash)) {
```

Comment:
What's wrong with `StringUtils` in this case - ?

Reply:
This was [imported from software.amazon.awssdk.utils.StringUtils](https://github.com/aws-powertools/powertools-lambda-java/pull/1513/files/0501b8fc9420e43252a9ca155e3906863c3d7af6#diff-567a5788e846a0897382a1883905c7cd891a7457ad733301fc41cfdad743eddbL42).
See the [cleanup commit](https://github.com/aws-powertools/powertools-lambda-java/pull/1513/commits/9f44ebf62b51a230a1907bbc33b055c49b29de68)